### PR TITLE
Dataviews: Add: Bulk actions to templates.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This PR implements bulk actions for templates.

The two bulk actions implemented are the revert template and the delete template.

## Testing

Create two user templates verify it is possible to bulk delete them.
Apply user changes to some default templates, and verify it is possible to bulk revert them.

## Screenshot

Screenshot 2023-11-28 at 21 06 49

### CODE REVIEW

**Comment by ntsekouras** [comment]
For me this is the use case where designers should give feedback. For example if an item has no actions at all, I guess we shouldn't show the selection checkbox. 

The most important thing IMO is how to handle if an item has actions, but we select two items that have different actions(one revert and the other delete).

File: packages/edit-site/src/components/page-templates/template-actions.js [hunk 1]
```diff
@@ -34,36 +34,59 @@ export function useResetTemplateAction() {
...
-							decodeEntities( template.title.rendered )
-						),
+						templates.length > 1
+							? sprintf(
+									/* translators: The number of items. */
+									__( '%s items reverted reverted.' ),
+									decodeEntities( templates.length )
+							  )
+							: sprintf(
+									/* translators: The template/part's name. */
```
Comment (afercia) [inline packages/edit-site/src/components/page-templates/template-actions.js:67]: For consistency with following translators comments, I'd suggest to use the same format:
/* translators: %s: The template or template part's name. */

File: packages/edit-site/src/components/page-templates/template-actions.js [hunk 2]
```diff
@@ -87,31 +110,83 @@ export const deleteTemplateAction = {
...
 			<VStack spacing="5">
 				<Text>
-					{ sprintf(
-						// translators: %s: The template or template part's title.
-						__( 'Are you sure you want to delete "%s"?' ),
-						decodeEntities( template.title.rendered )
-					) }
+					{ templates.length > 1
+						? sprintf(
+								// translators: %s: The template or template part's title.
```
Comment (afercia) [inline packages/edit-site/src/components/page-templates/template-actions.js:127]: This comment should refer to the number of items: `translators: The number of items.`

File: packages/edit-site/src/components/page-templates/template-actions.js [hunk 3]
```diff
@@ -34,36 +34,59 @@ export function useResetTemplateAction() {
...
 					createSuccessNotice(
-						sprintf(
-							/* translators: The template/part's name. */
-							__( '"%s" reverted.' ),
-							decodeEntities( template.title.rendered )
-						),
+						templates.length > 1
+							? sprintf(
+									/* translators: The number of items. */
+									__( '%s items reverted reverted.' ),
```
Comment (afercia) [inline packages/edit-site/src/components/page-templates/template-actions.js:63]: Typo: double 'reverted'.

**Comment by jameskoster** [comment]
Thanks for the ping, I'm very excited about this one!

> For example if an item has no actions at all, I guess we shouldn't show the selection checkbox.

It might be good to try showing the checkbox in the disabled state. On hover/focus is can show a tooltip explaining that there are no actions to apply to this record. 

> The most important thing IMO is how to handle if an item has actions, but we select two items that have different actions(one revert and the other delete).

Good question. Would it be expensive to show a count on the buttons? Something like:

Screenshot 2023-11-30 at 15 45 28

That can help the user understand that the chosen action cannot be applied to the full selection.

A couple of minor notes from initial testing:

* The checkbox in the table head should switch to the `indeterminate` state when there's a mixture of selected/deselected rows.
* The sticky pagination seems to be broken on this PR.
* The toolbar should stick at the bottom of the viewport, not the full list.

**Comment by afercia** [comment]
> * The toolbar should stick at the bottom of the viewport, not the full list.

Noting there's an ongoing discussion about the floating toolbar in the main 'bulk actions' PR, as it doesn't seem to be the most usable and accessible pattern. 

Aslo, quoting [this comment](https://github.com/WordPress/gutenberg/pull/56476#issuecomment-1838274582) from there:

> Let's please avoid floating / sticky / fixed UIs please, as they come with inherent usability and accessibility problems especially for keyboard users.

**Pull request closed without merging**